rust/azure_iot_operations_connector/src/file_mount_azure_device_registry.rs

```diff
@@ -1,4 +1,270 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
-//! Wrapper for Azure Device Registry Client that uses file mount to get names and create/delete notifications.
+//!Azure Device Registry Client that uses file mount to get names and create/delete notifications.
+
+use notify::RecommendedWatcher;
+use std::path::PathBuf;
+use std::sync::{Arc, Mutex};
+use std::time::Duration;
+use tokio::sync::mpsc::UnboundedReceiver;
+use tokio::sync::oneshot;
+
+const ADR_RESOURCES_NAME_MOUNT_PATH: &str = "/etc/akri/config/adr_resources_names";
+
+/// A client that interacts with the file mount
+///
+/// This client provides functionality to retrieve device names and handle
+/// create/delete notifications from the Azure Device Registry.
+#[allow(dead_code)]
+pub struct FileMountClient {
+    /// The path to the file mount used by the client.
+    pub mount_path: PathBuf,
+    /// A file watcher used to monitor changes in the file mount.
+    pub watcher: Arc<Mutex<RecommendedWatcher>>,
+}
+
+impl FileMountClient {
+    /// Creates a new instance of the `FileMountClient`.
+    ///
+    /// # Arguments
+    /// * `mount_path_env_var` - The environment variable containing the path to the file mount.
+    ///
+    /// # Returns
+    /// A `Result` containing the initialized `FileMountClient` or a `FileMountError`.
+    /// # Errors
+    /// Returns an error if the file mount cannot be accessed or if there is an issue with the watcher.
+    pub fn new() -> Result<Self, FileMountError> {
+        // read env vars here direclty without taking them, const at top of files
+        let mount_path = PathBuf::from(ADR_RESOURCES_NAME_MOUNT_PATH);
+        let watcher = notify::recommended_watcher(|_| {}).map_err(FileMountError::NotifyError)?;
+
+        Ok(Self {
+            mount_path,
+            watcher: Arc::new(Mutex::new(watcher)),
+        })
+    }
+
+    /// Gets names of all devices from the file mount.
+    ///
+    /// # Arguments
+    /// * `_timeout` - The duration to wait for the operation to complete.
+    ///
+    /// # Returns
+    /// A vector of [`DeviceEndpointRef`].
+    ///
+    /// # Errors
+    /// Returns an error if the file mount cannot be accessed or if there is an issue with the watcher.
+    pub fn get_device_names(
```

Comment:
```suggestion
    pub fn get_device_endpoint_names(
```
maybe update to this since we'll be getting both? Not sure if this feels entirely accurate either though